src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -377,8 +367,8 @@ private int countCurrentEC2Slaves(SlaveTemplate template) throws AmazonClientExc
             jenkinsServerUrl = jenkinsLocation.getUrl();
 
         if (jenkinsServerUrl == null) {
-            LOGGER.log(Level.WARNING, "No Jenkins server URL specified, using instance-identity instead");
-            jenkinsServerUrl = getInstanceIdentityEncodedPublicKey();
```

Comment:
not trying anymore to be smart for no good reason, just warn and **not** crash if serverUrl is unspecified.